src/components/NumericStepper/__tests__/NumericStepper.spec.tsx

```diff
@@ -0,0 +1,278 @@
+        await decreaseClick();
+
+        expect(onChangeMock).toBeCalledTimes(1);
+        expect(onChangeMock).toBeCalledWith(2);
+        expect(inputButton).toHaveAttribute('value', '2');
+        expect(onChangeMock).toHaveBeenCalledWith(2);
+
+        // Decrease by 2 (to equal to 0) and make sure that number of onChange calls equal to 2
+        await decreaseClick();
+
+        expect(onChangeMock).toBeCalledTimes(2);
+        expect(onChangeMock).toBeCalledWith(0);
+        expect(inputButton).toHaveAttribute('value', '0');
+        expect(onChangeMock).toHaveBeenCalledWith(0);
+    });
+
+    it('should react on stepUp click', async () => {
+        view = render(<NumericStepper onChange={onChangeMock} step={2} />);
+        const inputButton = screen.getByRole('spinbutton');
+
+        // Increase by 2 (to equal to 2) and make sure that number of onChange calls equal to 1
+        await increaseClick();
+
+        expect(onChangeMock).toBeCalledTimes(1);
+        expect(onChangeMock).toBeCalledWith(2);
+        expect(inputButton).toHaveAttribute('value', '2');
+        expect(onChangeMock).toHaveBeenCalledWith(2);
+
+        // Increase by 2 (to equal to 4) and make sure that number of onChange calls equal to 2
+        await increaseClick();
+
+        expect(onChangeMock).toBeCalledTimes(2);
+        expect(onChangeMock).toBeCalledWith(4);
+        expect(inputButton).toHaveAttribute('value', '4');
+        expect(onChangeMock).toHaveBeenCalledWith(4);
+    });
+
+    it('should disable buttons when limits are reached', async () => {
+        const user = userEvent.setup();
+        view = render(
+            <NumericStepper onChange={onChangeMock} minValue={1} maxValue={3} defaultValue={2} />
+        );
+        const inputButton = screen.getByRole('spinbutton');
+        const stepperButtons = screen.getAllByRole('button');
+
+        // Simulate two clicks on stepUp button
+        await increaseClick();
+
+        expect(inputButton).toHaveAttribute('value', '3');
+        expect(onChangeMock).toBeCalledTimes(1);
+        expect(stepperButtons[1]).toHaveAttribute('disabled');
+        expect(onChangeMock).toHaveBeenCalledWith(3);
+
+        // Simulate two clicks on stepDown button
+        await user.click(stepperButtons[0]);
+
+        expect(onChangeMock).toBeCalledTimes(2);
+
+        await user.click(stepperButtons[0]);
+
+        expect(onChangeMock).toBeCalledTimes(3);
+        expect(inputButton).toHaveAttribute('value', '1');
+        expect(stepperButtons[0]).toHaveAttribute('disabled');
+    });
+
+    it('should set bottom edge value if user enters value below allowed limit', async () => {
+        const user = userEvent.setup();
+        view = render(<NumericStepper onChange={onChangeMock} minValue={10} />);
+        const inputButton = screen.getByRole('spinbutton');
+
+        await user.type(inputButton, '9');
+
+        expect(inputButton).toHaveAttribute('value', '10');
```

Comment:
In the original implementation, the value was 9. I think it's wrong because the low value must be 10.

Reply:
I see 10 in the original too.